File: objc/Project4/Project4 WatchKit Extension/InterfaceController.m
//
//  InterfaceController.m
//  Project4 WatchKit Extension
//
//  Created by Jinwoo Kim on 1/24/21.
//

#import "InterfaceController.h"

@implementation InterfaceController

+ (NSArray<NSString *> *)currencies {
    static NSArray<NSString *> *currencies = nil;
    if (currencies == nil) {
        currencies = @[@"USD", @"AUD", @"CAD", @"CHF", @"CNY", @"EUR", @"GBP", @"HKD", @"JPY", @"SGD"];
    }
    return currencies;
}

+ (NSArray<NSString *> *)defaultCurrencies {
    static NSArray<NSString *> *defaultCurrencies = nil;
    if (defaultCurrencies == nil) {
        defaultCurrencies = @[@"USD", @"EUR"];
    }
    return defaultCurrencies;
}

+ (NSString *)selectedCurrenciesKey {
    static NSString *selectedCurrenciesKey = nil;
    if (selectedCurrenciesKey == nil) {
        selectedCurrenciesKey = @"SelectedCurrencies";
    }
    return selectedCurrenciesKey;
}

- (void)awakeWithContext:(id)context {
    [self setup];
    [super awakeWithContext:context];
    
    NSMutableArray<WKPickerItem *> *items = [@[] mutableCopy];
    
    for (NSString *currency in [[self class] currencies]) {
        WKPickerItem *item = [WKPickerItem new];
        item.title = currency;
        [items addObject:item];
    }
    
    [self.currencyPicker setItems:items];
}

- (void)setup {
    self.currentCurrency = @"USD";
    self.currentAmount = 500;
}

- (IBAction)amountChanged:(float)value {
    self.currentAmount = (int)value;
    [self.amountLabel setText:[NSString stringWithFormat:@"%d", self.currentAmount]];
}

- (IBAction)convertTapped {
    NSDictionary<NSString *, NSString *> *context = @{@"amount": [NSString stringWithFormat:@"%d", self.currentAmount],
                              @"base": self.currentCurrency};
    [WKInterfaceController reloadRootPageControllersWithNames:@[@"Results"]
                                                     contexts:@[context]
                                                  orientation:WKPageOrientationHorizontal
                                                    pageIndex:0];
}

- (IBAction)baseCurrencyChanged:(NSInteger)value {
    self.currentCurrency = [[self class] currencies][value];
}

@end

File: swift/Project4/Project4 WatchKit Extension/InterfaceController.swift
//
//  InterfaceController.swift
//  Project4 WatchKit Extension
//
//  Created by Jinwoo Kim on 1/23/21.
//

import WatchKit
import Foundation


class InterfaceController: WKInterfaceController {
    @IBOutlet weak var amountLabel: WKInterfaceLabel!
    @IBOutlet weak var amountSlider: WKInterfaceSlider!
    @IBOutlet weak var currencyPicker: WKInterfacePicker!
    
    static let currencies = ["USD", "AUD", "CAD", "CHF", "CNY", "EUR", "GBP", "HKD", "JPY", "SGD"]
    static let defaultCurrencies = ["USD", "EUR"]
    static let selectedCurrenciesKey = "SelectedCurrencies"
    var currentCurrency = "USD"
    var currentAmount = 500
    
    override func awake(withContext context: Any?) {
        super.awake(withContext: context)
        
        var items = [WKPickerItem]()
        
        for currency in Self.currencies {
            let item = WKPickerItem()
            item.title = currency
            items.append(item)
        }
        
        currencyPicker.setItems(items)
    }
    
    @IBAction func amountChanged(_ value: Float) {
        currentAmount = Int(value)
        amountLabel.setText(String(currentAmount))
    }
    
    @IBAction func convertTapped() {
        let context = ["amount": String(currentAmount), "base": currentCurrency]
        WKInterfaceController.reloadRootPageControllers(withNames: ["Results"], contexts: [context], orientation: .horizontal, pageIndex: 0)
    }
    
    @IBAction func baseCurrencyChanged(_ value: Int) {
        currentCurrency = Self.currencies[value]
    }
}
